lib/diversity/component.rb

```diff
@@ -56,30 +56,26 @@ def initialize(spec, options)
       @options = DEFAULT_OPTIONS.keep_merge(options)
       @logger = @options[:logger]
 
-      schema = JsonSchemaCache[
-                 MASTER_COMPONENT_SCHEMA,
-                 { validate_spec: @options[:validate_spec] }
-               ]
-
       # Handle both Hash and String specs
       @raw = (spec.is_a?(String) ? parse_config(spec) : spec)
 
-      validation = schema.validate(@raw)
-      raise Diversity::Exception, "Configuration is not valid:\n" + validation.join("\n"), caller if
-        validation.length > 0
+      if (@options[:validate_spec])
+        schema = JsonSchemaCache[
+          MASTER_COMPONENT_SCHEMA,
+          { validate_spec: @options[:validate_spec] }
+        ]
+
+        validation = schema.validate(@raw)
+        raise Diversity::Exception,
+              "Configuration is not valid:\n" + validation.join("\n"), caller if
+          validation.length > 0
+      end
 
-      @checksum = Digest::SHA1.hexdigest(dump)
+      @checksum = "#{name}:#{version}"
       @assets = {}
       populate(@raw)
     end
 
-    # Returns a JSON dump of the component configuration
-    #
-    # @return [String]
-    def dump
```

Comment:
Since @raw is not public, you're essentially removing the only way we can expose the component as JSON. Maybe we should replace this method with another one called to_json (that does not rely on pretty-printing), but I think it is a good idea to have such a method available. You **might** want to serialize a component (for caching etc).